packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -360,6 +366,50 @@ type IsFixedAwareUser = User & {
   organization: { slug: string };
 };
 
+const loadUsers = async (eventType: NewBookingEventType, dynamicUserList: string[], req: NextApiRequest) => {
+  try {
+    if (!eventType.id) {
+      if (!Array.isArray(dynamicUserList) || dynamicUserList.length === 0) {
+        throw new Error("dynamicUserList is not properly defined or empty.");
+      }
+
+      const users = await prisma.user.findMany({
+        where: {
+          username: { in: dynamicUserList },
+          organization: userOrgQuery(req.headers.host ? req.headers.host.replace(/^https?:\/\//, "") : ""),
+        },
+        select: {
+          ...userSelect.select,
+          credentials: {
+            select: credentialForCalendarServiceSelect,
+          },
+          metadata: true,
+        },
+      });
+
+      return users;
+    } else {
```

Comment:
suggestion: the `else` statement is not necessary here since we return in the `if` statement above, avoiding nesting whenever possible is always a good idea in my opinion

Reply:
The if/else statement checks for `eventType.id`. Dynamic events don't have an event type id. So this block is pretty much 
```
if (!eventType.id) {
    Get users for dynamic event
} else {
    Get users from normal event type
}
```

Reply:
Still is not needed since we're returning.

```
if (!eventType.id) {
    Get users for dynamic event
   return
} 

// this won't run if we have eventType.id
Get users from normal event type
return
```